src/main/java/io/airlift/slice/Slices.java

```diff
@@ -66,9 +71,13 @@ public static Slice ensureSize(Slice existingSlice, int minWritableBytes)
             }
         }
 
-        Slice newSlice = allocate(newCapacity);
-        newSlice.setBytes(0, existingSlice, 0, existingSlice.length());
-        return newSlice;
+        byte[] bytes = existingSlice.byteArray();
+        int offset = existingSlice.byteArrayOffset();
+        // copy of range can be more efficient because it avoids zeroing memory
+        byte[] copy = Arrays.copyOfRange(bytes, offset, offset + newCapacity);
+        // but if the slice is a view, we need to zero the end of the array
+        Arrays.fill(copy, newCapacity, copy.length, (byte) 0);
```

Comment:
This is wrong since `newCapacity == copy.length` and thus it actually does nothing. We only need to fill up to the number of extra bytes in the original array (which could be larger than the size of our copied slice). Something like this:
```java
int valid = existingSlice.length();
int extra = bytes.length - offset - valid;
Arrays.fill(copy, valid, min(valid + extra, copy.length), (byte) 0);
```

Reply:
It looks like `valid` can be eliminated because we're subtracting and then adding it, but that's clearly not right...